zetaclient/orchestrator/bootstrap.go

```diff
@@ -0,0 +1,414 @@
+	btcobserver "github.com/zeta-chain/zetacore/zetaclient/chains/bitcoin/observer"
+	"github.com/zeta-chain/zetacore/zetaclient/chains/bitcoin/rpc"
+	btcsigner "github.com/zeta-chain/zetacore/zetaclient/chains/bitcoin/signer"
+	evmobserver "github.com/zeta-chain/zetacore/zetaclient/chains/evm/observer"
+	evmsigner "github.com/zeta-chain/zetacore/zetaclient/chains/evm/signer"
+	"github.com/zeta-chain/zetacore/zetaclient/chains/interfaces"
+	solbserver "github.com/zeta-chain/zetacore/zetaclient/chains/solana/observer"
+	"github.com/zeta-chain/zetacore/zetaclient/config"
+	zctx "github.com/zeta-chain/zetacore/zetaclient/context"
+	"github.com/zeta-chain/zetacore/zetaclient/db"
+	"github.com/zeta-chain/zetacore/zetaclient/metrics"
+)
+
+// backwards compatibility
+const btcDatabaseFilename = "btc_chain_client"
+
+// CreateSignerMap creates a map of interfaces.ChainSigner (by chainID) for all chains in the config.
+// Note that signer construction failure for a chain does not prevent the creation of signers for other chains.
+func CreateSignerMap(
+	ctx context.Context,
+	tss interfaces.TSSSigner,
+	logger base.Logger,
+	ts *metrics.TelemetryServer,
+) (map[int64]interfaces.ChainSigner, error) {
+	signers := make(map[int64]interfaces.ChainSigner)
+	_, _, err := syncSignerMap(ctx, tss, logger, ts, &signers)
+	if err != nil {
+		return nil, err
+	}
+
+	return signers, nil
+}
+
+// syncSignerMap synchronizes the given signers map with the signers for all chains in the config.
+// This semantic is used to allow dynamic updates to the signers map.
+// Note that data race handling is the responsibility of the caller.
+func syncSignerMap(
+	ctx context.Context,
+	tss interfaces.TSSSigner,
+	logger base.Logger,
+	ts *metrics.TelemetryServer,
+	signers *map[int64]interfaces.ChainSigner,
+) (int, int, error) {
+	if signers == nil {
+		return 0, 0, errors.New("signers map is nil")
+	}
+
+	app, err := zctx.FromContext(ctx)
+	if err != nil {
+		return 0, 0, errors.Wrapf(err, "failed to get app context")
+	}
+
+	var (
+		added, removed int
+
+		presentChainIDs = make([]int64, 0)
+
+		onAfterAdd = func(chainID int64, _ interfaces.ChainSigner) {
+			logger.Std.Info().Msgf("Added signer for chain %d", chainID)
+			added++
+		}
+
+		addSigner = func(chainID int64, signer interfaces.ChainSigner) {
+			mapSet[int64, interfaces.ChainSigner](signers, chainID, signer, onAfterAdd)
+		}
+
+		onBeforeRemove = func(chainID int64, _ interfaces.ChainSigner) {
+			logger.Std.Info().Msgf("Removing signer for chain %d", chainID)
+			removed++
+		}
+	)
+
+	// EVM signers
+	for _, evmConfig := range app.Config().GetAllEVMConfigs() {
+		chainID := evmConfig.Chain.ChainId
+
+		if evmConfig.Chain.IsZetaChain() {
+			continue
+		}
+
+		evmChainParams, found := app.GetEVMChainParams(chainID)
+		switch {
+		case !found:
+			logger.Std.Warn().Msgf("Unable to find chain params for EVM chain %d", chainID)
+			continue
+		case !evmChainParams.IsSupported:
```

Comment:
The `IsSupported` flag is supposed to be `true` because it is fetched [from the query GetSupportedChains](https://github.com/zeta-chain/node/blob/feat%2Fzetaclient-runtime-chains/zetaclient/zetacore/client.go#L402). If we do want to double check it, we can just check it right in `client.go`.

Reply:
I prefer to keep this because this code is not aware of the source of ChainParams.

Reply:
If we really want to enforce something, I think enforcing it in the very beginning source could stop spreading checking in all downstream code. Just my opinion.